quinn-proto/src/connection/mod.rs

```diff
@@ -2833,10 +2976,55 @@ impl Connection {
             self.stats.frame_tx.ping += 1;
         }
 
+        // IMMEDIATE_ACK
+        if mem::replace(&mut space.immediate_ack_pending, false) {
+            trace!("IMMEDIATE_ACK");
+            buf.write(frame::Type::IMMEDIATE_ACK);
+            sent.non_retransmits = true;
+            self.stats.frame_tx.immediate_ack += 1;
+        }
+
         // ACK
         if space.pending_acks.can_send() {
             debug_assert!(!space.pending_acks.ranges().is_empty());
-            Self::populate_acks(self.receiving_ecn, &mut sent, space, buf, &mut self.stats);
+            Self::populate_acks(
+                now,
+                self.receiving_ecn,
+                &mut sent,
+                space,
+                buf,
+                &mut self.stats,
+            );
+        }
+
+        // ACK_FREQUENCY
+        if let Some(sequence_number) = space.pending.ack_frequency.take() {
+            trace!("ACK_FREQUENCY");
+
+            // Safe to unwrap because these are always provided when ACK frequency is enabled
+            let config = self.config.ack_frequency_config.as_ref().unwrap();
+            let min_ack_delay_micros = self.peer_params.min_ack_delay.unwrap().0;
+
+            // Ensure the delay is within bounds to avoid a PROTOCOL_VIOLATION error
+            let max_ack_delay = self.ack_frequency.candidate_max_ack_delay(config);
+            let max_ack_delay_micros = max_ack_delay
+                .as_micros()
+                .try_into()
+                .unwrap_or(u64::MAX)
```

Comment:
`u64::MAX` is out of range for varints, so this will lead to a panic. Should we clamp to `VarInt::MAX.into_inner()`?